AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/helper/IntentBuilder.java

```diff
@@ -135,16 +139,18 @@ public JInvocation getSuperPutExtraInvocation(TypeMirror elementType, JVar extra
 		if (elementType.getKind() == TypeKind.DECLARED) {
 			Elements elementUtils = environment.getProcessingEnvironment().getElementUtils();
 			TypeMirror parcelableType = elementUtils.getTypeElement(PARCELABLE).asType();
-			if (!typeUtils.isSubtype(elementType, parcelableType)) {
-				TypeMirror stringType = elementUtils.getTypeElement(STRING).asType();
-				if (!typeUtils.isSubtype(elementType, stringType)) {
-					extraParameterArg = cast(environment.getClasses().SERIALIZABLE, extraParameterArg);
-				}
-			} else {
+			if (typeUtils.isSubtype(elementType, parcelableType)) {
 				TypeMirror serializableType = elementUtils.getTypeElement(SERIALIZABLE).asType();
 				if (typeUtils.isSubtype(elementType, serializableType)) {
 					extraParameterArg = cast(environment.getClasses().PARCELABLE, extraParameterArg);
 				}
+			} else if (!BundleHelper.METHOD_SUFFIX_BY_TYPE_NAME.containsKey(elementType.toString()) && annotationHelper.isParcelType(elementType)) {
```

Comment:
Totally agree, the current code is just too complicated...

I also thought about creating a provider thing, which we discussed in your
original PR.
That way we could completely split off Parceler support to a plugin,
instead of adding Parceler-related
code directly to AA.

Your code is very clean and robust, congratulations!